pkg/app/interactive.go

```diff
@@ -0,0 +1,288 @@
+		"python",
+		"r",
+		"julia",
+	},
+}
+
+var PythonPackageChoice = input{
+	prompt:    "Choose your python packages",
+	inputType: MULTIPLE_SELECT,
+	label:     LabelPythonPackage,
+	options: []string{
+		"numpy",
+		"tensorflow",
+	},
+}
+
+var RPackageChoice = input{
+	prompt:    "Choose your r packages",
+	inputType: MULTIPLE_SELECT,
+	label:     LabelRPackage,
+	options: []string{
+		"remotes",
+		"rlang",
+	},
+}
+
+var CudaChoice = input{
+	prompt:    "Include Cuda?",
+	inputType: SINGLE_SELECT,
+	label:     LabelCudaChoice,
+	options: []string{
+		"Yes",
+		"No",
+	},
+}
+
+var CudaVersionChoice = input{
+	prompt:    "Choose a cuda version",
+	inputType: SINGLE_SELECT,
+	label:     LabelCuda,
+	options: []string{
+		"11.6.2",
+		"11.3.1",
+		"11.2.2",
+	},
+}
+
+var JupyterChoice = input{
+	prompt:    "Include Jupyter?",
+	inputType: SINGLE_SELECT,
+	label:     LabelJupyterChoice,
+	options: []string{
+		"Yes",
+		"No",
+	},
+}
+
+func InitModel(input input) model {
+	return model{
+		input:    input,
+		step:     0,
+		selected: make(map[int]struct{}),
+		exit:     false,
+	}
+}
+
+func (m model) View() string {
+	s := m.input.prompt
+
+	switch m.input.inputType {
+	case SINGLE_SELECT, MULTIPLE_SELECT:
+		s += m.renderMultipleChoice()
+
+	case INPUT:
+		// TODO: implement input if needed
+
+	}
+
+	s += "\nPress q to quit. "
+	if m.input.inputType == MULTIPLE_SELECT {
+		s += "Press space to select"
+	}
+
+	return s
+}
+
+func (m model) Init() tea.Cmd {
+	return nil
+}
+
+func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	choices := m.input.options
+
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c", "q":
+			m.exit = true
+			return m, tea.Quit
+		case "up", "k":
+			if m.cursor > 0 {
+				m.cursor--
+			}
+		case "down", "j":
+			if m.cursor < len(choices)-1 {
+				m.cursor++
+			}
+		case " ":
+			if m.input.inputType == MULTIPLE_SELECT {
+				_, ok := m.selected[m.cursor]
+				if ok {
+					delete(m.selected, m.cursor)
+				} else {
+					m.selected[m.cursor] = struct{}{}
+				}
+			}
+		case "enter":
+			if m.input.inputType == SINGLE_SELECT {
+				m.selected[m.cursor] = struct{}{}
+			}
+			selectionMap[m.input.label] = []string{choices[m.cursor]}
+			m.addSelection()
+			return m, tea.Quit
+		}
+	}
+
+	return m, nil
+}
+
+func (m model) addSelection() {
+	selectionMap[m.input.label] = []string{}
+	for i := range m.selected {
+		selectionMap[m.input.label] = append(selectionMap[m.input.label], m.input.options[i])
+	}
+}
+
+func startQuestion(input input) {
+	p := tea.NewProgram(InitModel(input))
+	m, err := p.Run()
+	if m.(model).exit {
+		os.Exit(0)
+	}
+	if err != nil {
+		fmt.Printf("There was an error generating build.envd: %v", err)
+		os.Exit(1)
+	}
+}
+func generateFile(clicontext *cli.Context) error {
+	var buf bytes.Buffer
+	buf.WriteString("def build():\n")
+	buf.WriteString(fmt.Sprintf("    base(os=\"ubuntu20.04\", language=\"%s\")\n", selectionMap[LabelLanguage][0]))
```

Comment:
All the 4-space in the `fmt.Sprintf` should use a const.

Reply:
Ok